src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/BinaryFormatterUnsafeDeserializeAnalyzer.cs

```diff
@@ -39,71 +38,81 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
-            context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressions, SyntaxKind.SimpleMemberAccessExpression);
+            context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressions, CodeAnalysis.CSharp.SyntaxKind.SimpleMemberAccessExpression);
+            context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressions, CodeAnalysis.VisualBasic.SyntaxKind.SimpleMemberAccessExpression);
         }
 
         private void AnalyzeMemberAccessExpressions(SyntaxNodeAnalysisContext context)
         {
-            var memberAccessExpression = (MemberAccessExpressionSyntax)context.Node;
-
-            // If the accessed member isn't named "UnsafeDeserialize" bail out
-            if (!TargetMember.Equals(memberAccessExpression.Name.ToString(), StringComparison.Ordinal))
+            var isMemberAccessExpresion = GeneralMemberAccessExpression.TryParse(context.Node, out var expression);
```

Comment:
Nit-pick: This can be combined -

```
if (!GeneralMemberAccessExpression(context.Node, out var expression) || expression is null)
{
    return;
}
```